core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/IoUtil.java

```diff
@@ -134,6 +141,51 @@ public static InputStream getResource(String resource) throws IOException {
         return is;
     }
 
+    /**
+     * Provides streams to all resources matching {@code filter} criteria.
+     * Method search inside archive if context represents a jar file.
+     *
+     * @param context
+     *      the current context
+     * @throws IOException
+     */
+    public static void getResources(Class context, Pattern filter, Consumer<InputStream> consumer) throws IOException {
```

Comment:
Divided into 2 methods: 
* finding resources inside zip archive (moved to ZipUtils)
* finding resources inside directory (moved to local usage)
Test added.